mwdb/web/src/commons/ui/ErrorBoundary.jsx

```diff
@@ -23,16 +25,46 @@ export function getErrorMessage(error) {
 export function Alert(props) {
     if (props.error) {
         return (
-            <div className="alert alert-danger">
-                {getErrorMessage(props.error)}
-            </div>
+            <AlertMessage
+                type={"danger"}
+                message={getErrorMessage(props.error)}
+            />
         );
-    } else if (props.success) {
-        return <div className="alert alert-success">{props.success}</div>;
-    } else if (props.warning) {
-        return <div className="alert alert-warning">{props.warning}</div>;
     }
-    return <div />;
+    if (props.success) {
+        return <AlertMessage type={"success"} message={props.success} />;
+    }
+    if (props.warning) {
+        return <AlertMessage type={"warning"} message={props.warning} />;
+    }
+    return <></>;
+}
+
+function AlertMessage(props) {
+    const { type, message } = props;
+    const [isVisible, setIsVisible] = useState(true);
+
+    useEffect(() => {
+        setIsVisible(true);
+    }, [props]);
```

Comment:
If you want to set it to visible on every render, I would just omit dependency argument in useEffect. Otherwise, use `[type, message]`.

Right now, useEffect fires on `props` object reference change instead of change of actual props

Reply:
The solution you propose is not entirely desirable for me, because:

I don't want this window to be visible after each render, because then closing it would make no sense (after typing 1 letter, it would appear again).
If I track changes directly on `[type, message]`, in the case of closing this window and calling the same error again, the window would never show up again (for example, in the case of repeated incorrect login attempts). That's why I'm observing all `props` that are objects, and in case of their new assignment, creating a copy of that object (for example, after submit login form), this error will be visible again. That's why I added a `memo` wrapper to the `ViewAlert` component.